src/main/java/com/company/KafkaProducerTest.java

```diff
@@ -18,19 +18,33 @@
 
 package com.company;
 
-import java.io.File;
-import java.io.FileReader;
-import java.io.IOException;
+import java.io.*;
 import java.util.Properties;
 
+import org.apache.commons.cli.*;
 import org.apache.hadoop.yarn.logaggregation.AggregatedLogFormat;
 import org.apache.kafka.clients.producer.KafkaProducer;
 import org.apache.kafka.clients.producer.Producer;
 import org.apache.kafka.clients.producer.ProducerRecord;
-import java.io.BufferedReader;
 
-public class  KafkaProducerTest {
-  public static void main(String[] args) {
+public class KafkaProducerTest {
+  public static void main(String[] args) throws Exception {
+
+    Options options = new Options();
+    Options topic = options.addOption("topic", true, "Configuring the topic");
+    Options fileName = options.addOption("fileName", true, "Configuring the file name");
+    Options brokerList = options.addOption("brokerList", true, "Configuring the broker list");
+
+    CommandLineParser parser = new DefaultParser();
+
+    // parse the command line arguments
+    CommandLine line = parser.parse(options, args);
+    CommandLine commandLine = null;
+    args = commandLine.getArgs();
+    if (args.length == 0) {
```

Comment:
I think this just gets "left over" args (https://commons.apache.org/proper/commons-cli/apidocs/org/apache/commons/cli/CommandLine.html#getArgs--) after you have parsed the options above. As such I think you can delete this.